spec/controllers/incomes_controller_spec.rb

```diff
@@ -0,0 +1,138 @@
+require "rails_helper"
+
+describe IncomesController do
+  context "GET #new" do
+    context "when invalid params" do
+      it "won't create a new income" do
+        create_current_user_with_budget
+        income_params = { name: "Salary", amount: 500, frequency: "" }
+
+        expect do
+          post :create, params: { income: income_params }
+        end.to change(Income, :count).by(0)
+      end
+
+      it "renders new" do
+        create_current_user_with_budget
+        income_params = { name: "Salary", amount: 500, frequency: "" }
+
+        post :create, params: { income: income_params }
+
+        expect(response).to render_template(:new)
+      end
+
+      it "sets the flash[:alert]" do
+        create_current_user_with_budget
+        income_params = { name: "Salary", amount: 500, frequency: "" }
+
+        post :create, params: { income: income_params }
+
+        expect(flash[:alert]).
+          to match I18n.
+          t("flash.actions.create.alert", resource_name: "Income")
```

Comment:
A suggestion to change the format to be:
```
        expect(flash[:alert]).to match(
          I18n.t("flash.actions.create.alert", resource_name: "Income"),
        )
```

Reply:
Done in sha: ea7bed2